keyvault/client/secrets/README.md

```diff
@@ -0,0 +1,252 @@
+        .build();
+```
+
+## Key concepts
+
+### Secret
+
+A secret is the fundamental resource within an Azure KeyVault. In its simplest form it is a name and a value. However, there are additional properties such as the modifiable content type and tags fields that allow the value to be interpreted or associated in different ways.
+The expires property of a Secret provides a way to specify a UTC time at which it will no loger be active. The notBefore property of a Secret allows to specify a UTC time after which secret will be active.
+
+### Secret Client
+
+The client performs the interactions with the Azure Key Vault service, getting, setting, updating, deleting, and listing secrets. An asynchronous, `SecretAsyncClient`, and synchronous, `SecretClient`, client exists in the SDK allowing for selection of a client based on an application's use case.
+
+## Sync API Examples
+
+The following sections provide several code snippets covering some of the most common Azure Key Vault Secret Service tasks, including:
+- [Create a Secret](#create-a-Secret)
+- [Retrieve a Secret](#retrieve-a-Secret)
+- [Update an existing Secret](#update-an-existing-Secret)
+- [Delete a Secret](#delete-a-Secret)
+
+### Create a Secret
+
+Create a Secret to be stored in the Azure Key Vault. There are two ways to store a Secret:
+- setSecret creates a new secret in the key vault. if the secret with name already exists then a new version of the secret is created.
+```Java
+SecretClient secretClient = SecretClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+Secret secret = secretClient.setSecret("secret_name", "secret_value").value();
+System.out.printf("Secret is created with name %s and value %s \n", secret.name(), secret.value());
+
+```
+
+### Retrieve a Secret
+
+Retrieve a previously stored Secret by calling getSecret.
+```Java
+SecretClient secretClient = SecretClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+secretClient.setSecret("secret_name", "secret_value");
+Secret secret = secretClient.getSecret("secret_name").value();
+System.out.printf("Secret is returned with name %s and value %s \n", secret.name(), secret.value());
+```
+
+### Update an existing Secret
+
+Update an existing Secret by calling updateSecret.
+```Java
+SecretClient secretClient = SecretClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+Secret secret = secretClient.setSecret("secret_name", "secret_value").value();
+secret.expires(OffsetDateTime.now().plusDays(30));
+SecretBase updatedSecret = secretClient.updateSecret(secret).value();
+System.out.printf("Secret's updated expiry time %s \n", updatedSecret.expires().toString());
+```
+
+### Delete a Secret
+
+Delete an existing Secret by calling deleteSecret.
```

Comment:
should we add some information here about the difference between delete and purge? or link to documentation about it?

Reply:
Not sure if we should get into soft-delete logic here.
We'll need to add an example of Purge for that.